File: gramarogya_frontend/src/services/medicineFollowUpService.js
import axiosClient from "../api/axiosClient";

const medicineFollowUpService = {

  // Create medicine follow-up
  createFollowUp: async (data) => {
    const response = await axiosClient.post(
      "/medicine-followups",
      data
    );
    return response.data;
  },

  // Get follow-ups by beneficiary
  getFollowUpsByBeneficiary: async (beneficiaryId) => {
    const response = await axiosClient.get(
      `/medicine-followups/beneficiary/${beneficiaryId}`
    );
    return response.data;
  },

  // Get follow-ups by visit
  getFollowUpsByVisit: async (visitId) => {
    const response = await axiosClient.get(
      `/medicine-followups/visit/${visitId}`
    );
    return response.data;
  },

  // Get follow-up by ID
  getFollowUpById: async (id) => {
    const response = await axiosClient.get(
      `/medicine-followups/${id}`
    );
    return response.data;
  },
};

export default medicineFollowUpService;

File: gramarogya-backend/src/main/java/com/gramarogya/gramarogya_backend/service/medicine/MedicineFollowUpService.java
package com.gramarogya.gramarogya_backend.service.medicine;

import com.gramarogya.gramarogya_backend.dto.medicine.CreateMedicineFollowUpRequestDto;
import com.gramarogya.gramarogya_backend.dto.medicine.MedicineFollowUpResponseDto;
import org.springframework.security.core.Authentication;

import java.util.List;

public interface MedicineFollowUpService {

    MedicineFollowUpResponseDto createFollowUp(
            CreateMedicineFollowUpRequestDto request,
            Authentication authentication
    );

    List<MedicineFollowUpResponseDto> getFollowUpsByBeneficiary(
            String beneficiaryId,
            Authentication authentication
    );

    List<MedicineFollowUpResponseDto> getFollowUpsByVisit(
            String visitId,
            Authentication authentication
    );

    MedicineFollowUpResponseDto getFollowUpById(
            String id,
            Authentication authentication
    );
}